sympy/simplify/combsimp.py

```diff
@@ -217,6 +219,232 @@ def explicate(p):
                 denom_gammas.extend(l)
             elif isg is False:
                 denom_others.extend(l)
+        # =========== level >= 2 work: factor absorbtion =========
+        if level >= 2:
```

Comment:
@jksuom , I have removed two functions from rule_gamma and changed one function for factorial purpose only . I have two doubts:
1) In the new factorial_factor function what does level 0,1 and 3 denote.
2) what does function explicate exactly do ?
Please explain , thanks.